### PR TITLE
#143 - Add average rating as item field

### PR DESCRIPTION
**Pull request by tmachows**

### CODE REVIEW

File: src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import org.bson.types.ObjectId;
+
+import java.util.Set;
+
+@RequiredArgsConstructor
+@Data
+public class ItemIdAndRatings {
+    private final ObjectId itemId;
+    private final Set<Rating> ratings;
+}
```
Comment (tmachows) [inline src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: @pkruczek Is that like model you were talking about? ;p  
It is now used for `ItemOnRatingChangeUpdater` as I needed somehow to pass all necessary data in one object to it.

Comment (pkruczek) [reply src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: Yes it's exactly what I meant :D

File: src/main/java/com/beerjournal/breweriana/item/rating/persistence/RatingRepository.java [hunk 2]
```diff
@@ -48,6 +54,15 @@ public Rating delete(ObjectId ratingId) {
...
                 .orElseThrow(() -> new BeerJournalException(ErrorInfo.RATING_NOT_FOUND));
         ratingCrudRepository.delete(ratingId);
+
+        notifyUpdate(ratingToDelete.getItemId());
         return ratingToDelete;
     }
+
+    private void notifyUpdate(ObjectId itemId) {
+        ratingUpdateListeners.forEach(listener -> {
+            Set<Rating> itemRatings = ratingCrudRepository.findAllByItemId(itemId);
```
Comment (pkruczek) [inline src/main/java/com/beerjournal/breweriana/item/rating/persistence/RatingRepository.java]: You don't need to fetch ratings in forEach. Move it outside the loop

File: src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.beerjournal.breweriana.item.rating.persistence.Rating;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.bson.types.ObjectId;
+
+import java.util.Set;
+
+@RequiredArgsConstructor
+@Data
+public class ItemIdAndRatings {
```
Comment (pkruczek) [inline src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: Maybe `ItemRatings` ?

File: src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.beerjournal.breweriana.item.rating.persistence.Rating;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.bson.types.ObjectId;
+
+import java.util.Set;
+
+@RequiredArgsConstructor
+@Data
```
Comment (pkruczek) [inline src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: Maybe `@Data(staticConstructor = "of")`

Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: Does it have any particular advantage besides one less annotation? I think this is quite simple case when we don't need factory method very much.
I am even willing to change it but I'd like to know the benefits well. ;)

File: src/main/java/com/beerjournal/breweriana/item/rating/persistence/RatingRepository.java (hunk 2 above)
Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/rating/persistence/RatingRepository.java]: Good catch, updated.

File: src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java (hunk 3 above)
Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: Updated.

File: src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java (hunk 4 above)
Comment (pkruczek) [reply src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: In this case there is no very important reason for creating static factory method, but all our model classes have such methods. For me it makes class similar to Scala case class and I think constructors are old-fashioned :D

Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/rating/ItemIdAndRatings.java]: I thought you didn't like Scala :P Updated.

**Review by pkruczek — APPROVED** [review]

**Pull request merged**